### ISSUE TITLE
Cannot access static files via Reactor Netty!

### ISSUE THREAD

**Issue by MuhammedSenussi** [bclozel/spring-boot-web-reactive#54]
Hi,

I got an exception while requesting static file using Netty server:

```
<dependency>
    <groupId>org.springframework.boot.experimental</groupId>
    <artifactId>spring-boot-starter-web-reactive</artifactId>
    <exclusions>
        <exclusion>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-tomcat</artifactId>
        </exclusion>
    </exclusions>
</dependency>
<dependency>
    <groupId>io.projectreactor.ipc</groupId>
    <artifactId>reactor-netty</artifactId>
</dependency>
```

Test:
```
@Test
	public void staticResources() throws Exception {
		ClientRequest<Void> request = ClientRequest
				.GET("http://localhost:{port}/static/index.txt", this.port)
				.accept(MediaType.TEXT_PLAIN).build();

		Mono<String> result = this.webClient
				.exchange(request)
				.then(response -> response.body(toMono(String.class)));

		ScriptedSubscriber.<String>create()
				.consumeNextWith(content -> {
					assertThat(content).contains("Hello World");
				})
				.expectComplete()
				.verify(result);
	}
```

The Error:

```
2016-12-03 14:56:05.910 ERROR 6075 --- [-server-epoll-6] o.s.w.s.h.ExceptionHandlingWebHandler    : Could not complete request

io.netty.handler.codec.EncoderException: java.lang.IllegalStateException: unexpected message type: DefaultFileRegion
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:106) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.CombinedChannelDuplexHandler.write(CombinedChannelDuplexHandler.java:345) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:749) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:741) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:827) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:734) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.ChannelDuplexHandler.write(ChannelDuplexHandler.java:106) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.handler.codec.http.HttpServerKeepAliveHandler.write(HttpServerKeepAliveHandler.java:87) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:749) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:741) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:827) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:734) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.ChannelDuplexHandler.write(ChannelDuplexHandler.java:106) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:749) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:812) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:825) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:805) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:842) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1032) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:296) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at reactor.ipc.netty.channel.ChannelOperations.lambda$sendFile$3(ChannelOperations.java:363) ~[reactor-netty-0.6.0.BUILD-SNAPSHOT.jar:na]
	at reactor.core.publisher.MonoUsing.subscribe(MonoUsing.java:83) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatIterable$ConcatIterableSubscriber.onComplete(FluxConcatIterable.java:152) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatIterable.subscribe(FluxConcatIterable.java:69) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoNext.subscribe(MonoNext.java:41) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.drain(FluxConcatMap.java:397) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.onSubscribe(FluxConcatMap.java:189) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxTake$TakeFuseableSubscriber.onSubscribe(FluxTake.java:414) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxJust.subscribe(FluxJust.java:69) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxTakeFuseable.subscribe(FluxTakeFuseable.java:54) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap.subscribe(FluxConcatMap.java:113) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoNext.subscribe(MonoNext.java:41) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenMap$MonoThenApplyMain.onNext(MonoThenMap.java:132) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:68) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:942) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenMap$MonoThenApplyMain.onNext(MonoThenMap.java:126) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoNext$NextSubscriber.onNext(MonoNext.java:78) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.drain(FluxConcatMap.java:380) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.onSubscribe(FluxConcatMap.java:189) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:93) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:56) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap.subscribe(FluxConcatMap.java:113) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoNext.subscribe(MonoNext.java:41) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenMap.subscribe(MonoThenMap.java:55) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoOtherwiseIfEmpty.subscribe(MonoOtherwiseIfEmpty.java:44) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenMap.subscribe(MonoThenMap.java:55) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenMap.subscribe(MonoThenMap.java:55) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenMap$MonoThenApplyMain.onNext(MonoThenMap.java:132) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:68) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoNext$NextSubscriber.onNext(MonoNext.java:78) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.innerNext(FluxConcatMap.java:258) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap$ConcatMapInner.onNext(FluxConcatMap.java:740) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:124) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:1565) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.request(FluxMapFuseable.java:180) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:1282) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:1245) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onSubscribe(FluxMapFuseable.java:94) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:54) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoMapFuseable.subscribe(MonoMapFuseable.java:64) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.drain(FluxConcatMap.java:397) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.onSubscribe(FluxConcatMap.java:189) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:93) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:56) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.FluxConcatMap.subscribe(FluxConcatMap.java:113) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoNext.subscribe(MonoNext.java:41) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoOtherwiseIfEmpty.subscribe(MonoOtherwiseIfEmpty.java:44) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenMap.subscribe(MonoThenMap.java:55) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenMap.subscribe(MonoThenMap.java:55) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoOtherwise.subscribe(MonoOtherwise.java:44) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoOtherwise.subscribe(MonoOtherwise.java:44) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoOtherwise.subscribe(MonoOtherwise.java:44) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenIgnore$MonoThenIgnoreMain.drain(MonoThenIgnore.java:166) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoThenIgnore.subscribe(MonoThenIgnore.java:54) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoOtherwise.subscribe(MonoOtherwise.java:44) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoPeek.subscribe(MonoPeek.java:85) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:63) ~[reactor-core-3.0.3.RELEASE.jar:na]
	at reactor.ipc.netty.channel.ChannelOperations.applyHandler(ChannelOperations.java:535) ~[reactor-netty-0.6.0.BUILD-SNAPSHOT.jar:na]
	at reactor.ipc.netty.http.server.HttpServerOperations.onInboundNext(HttpServerOperations.java:414) ~[reactor-netty-0.6.0.BUILD-SNAPSHOT.jar:na]
	at reactor.ipc.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:70) ~[reactor-netty-0.6.0.BUILD-SNAPSHOT.jar:na]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.handler.codec.http.HttpServerKeepAliveHandler.channelRead(HttpServerKeepAliveHandler.java:64) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:1018) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:402) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:307) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873) ~[netty-common-4.1.6.Final.jar:4.1.6.Final]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_111]
Caused by: java.lang.IllegalStateException: unexpected message type: DefaultFileRegion
	at io.netty.handler.codec.http.HttpObjectEncoder.encode(HttpObjectEncoder.java:97) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:88) ~[netty-all-4.1.6.Final.jar:4.1.6.Final]
	... 107 common frames omitted

```

**Comment by bclozel** [comment]
Thanks for raising this issue!
For reference, this has been raised as well with [SPR-14975](https://jira.spring.io/browse/SPR-14975).

**Comment by bclozel** [comment]
Fixed in Spring Framework as of 5.0.M4
